---
title: "Class 1"
subtitle: "Meet & Greet + PC Set up! - ABEM 22-23"
author: "Dr. Giuliano Colosimo & Giulia Maiello"
footer:  "[https://giulianocolosimo.github.io/abem_22-23/](https://giulianocolosimo.github.io/abem_22-23/)"
logo: ../images/background.jpg
format: 
  revealjs: 
    theme: slides.scss
    multiplex: true
    transition: fade
    slide-number: true
editor: visual
execute:
  freeze: auto
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(countdown)
library("plotrix")
library("latticeExtra")
library("formattable")
library("lubridate")
```


# Welcome

## Your teachers

::: columns

::: {.column width="50%"}
![Dr. Giuliano Colosimo](../images/class1/giuliano_iguana.jpg){fig-alt="Dr. Giuliano Colosimo and its iguana friend" fig-align="center"}
:::

:::{.column width="50%"}
- Research Associate working @ University of Rome "Tor Vergata" 
- I was a PostDoc for 5 years with the San Diego Zoo Wildlife Association
- I got a PhD in Conservation Genetics @ Mississippi State University
- giuliano.colosimo@uniroma2.it
:::
:::

## 

::: columns
::: {.column width="50%"}
![Giulia Maiello](../images/class1/giulia.jpg){fig-alt="Giulia Maiello" fig-align="center"}
:::

:::{.column width="50%"}
- PhD Student
- gmaiello93@gmail.com
:::
:::

# Bioinformatics

## What is Bioinformatics?

:::{.callout-note icon=false}
## From [Wikipedia](https://en.wikipedia.org/wiki/Bioinformatics)
Bioinformatics is an interdisciplinary field that develops methods and software tools for understanding biological data, in particular when the data sets are large and complex. As an interdisciplinary field of science, bioinformatics combines biology, chemistry, physics, computer science, information engineering, mathematics and statistics to analyze and interpret the biological data. Bioinformatics has been used for in silico analyses of biological queries using computational and statistical techniques.
:::

:::{.callout-important}
Bioinformaticians deal with a variety of different type of data! We are not bioinformaticians, we are evolutionary biologists and we will be dealing with the data of greatest interest to us: nucleotide sequence data.
:::


## Why Bioinformatics

Back in the days, biologists could analyze their small and manageable data on their personal computers. [Sanger Sequecing](https://en.wikipedia.org/wiki/Sanger_sequencing) produced anywhere between 50 and 1000 nucleotides at a time to be analysed on a regular computer.

Since 1994 scientist have begun developing new technologies to parallelize DNA sequencing and produce high throughput data.

As large dataset become available, biologists need to harness the tools to design meaningful experiments to produce data and analyse them.

##

```{r seq_cost,echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Drop of sequencing cost for Genomes and Nucleotides Data from [https://www.genome.gov/about-genomics/fact-sheets/DNA-Sequencing-Costs-Data](https://www.genome.gov/about-genomics/fact-sheets/DNA-Sequencing-Costs-Data)"}
# data downloaded from https://www.genome.gov/about-genomics/fact-sheets/DNA-Sequencing-Costs-Data

seq_cost <- read.csv("~/Documents/websites/abem_22-23/data/Sequencing_Cost_Data_Table_Aug2021.csv",
                     header = T)

seq_cost <- seq_cost[1:75, 1:3]

seq_cost$Cost.per.Mb <- currency(seq_cost$Cost.per.Mb, "$")
seq_cost$Cost.per.Genome <- currency(seq_cost$Cost.per.Genome, "$")

seq_cost$Date <- sub(pattern = "-", "", seq_cost$Date, fixed = T)
seq_cost$Date <- paste("01", seq_cost$Date, sep = "")
seq_cost$Date <- as.Date(seq_cost$Date, format = "%d%b%y")

seq_cost$scaledMB <- scale(seq_cost$Cost.per.Mb)
seq_cost$scaledGen <- scale(seq_cost$Cost.per.Genome)

par(mar = rep(5, 4))
plot(seq_cost$Date, seq_cost$scaledGen, type = "l", lwd = 20, col = "blue",
     ylab = "", xlab = "Time", ylim = c(-.5, 5),
     xlim = c(as.Date("2000-01-01"), as.Date("2025-01-01")),
     bty="n", yaxt = "n")
lines(seq_cost$Date, seq_cost$scaledMB, type = "l", lwd = 5, col = "red")
axis(side = 2, at = c(-.5, 2.25, 5), labels = c("$ 689", "$ 50Mln", "$ 100Mln"), las = 2, col = "blue", col.ticks = "blue", col.axis = "blue",
     lwd = 3)
axis(side = 4, at = c(-.5, 2.25, 5), labels = c("$ 0.008", "$ 2.5K", "$ 5K"), las = 2, col = "red", col.ticks = "red", col.axis = "red",
     lwd = 3)
legend("top", col = c("blue", "red"), legend = c("Cost per Genome", "Cost per Mb"),
       lty = 1, lwd = c(10, 5), bty = "n")
```

# Setting up your computer

To visualize and analyse data and to produce reports on all the activities of this course we will be using **R** and **RStudio**. These are free cross-platform software. 

- Visit the **R** (https://cran.r-project.org) page and download the **R** version compatible to your OS. Install the software following the instructions.

- Visit the **RStudio** (https://www.rstudio.com/products/rstudio/download/) page and download the **RStudio** Desktop version compatible with your OS. Install the software following the instructions.
