---
title: "Class 1"
subtitle: "Meet & Greet + PC Set up! - ABEM 22-23"
author: "Dr. Giuliano Colosimo & Giulia Maiello"
footer:  "[https://giulianocolosimo.github.io/abem_22-23/](https://giulianocolosimo.github.io/abem_22-23/)"
logo: ../images/background.jpg
format: 
  revealjs: 
    theme: slides.scss
    multiplex: true
    transition: fade
    slide-number: true
    preview-links: true
    chalkboard: 
      boardmarker-width: 5
editor: visual
from: markdown+emoji
execute:
  freeze: auto
editor_options: 
  chunk_output_type: console
bibliography: /Users/giulianocolosimo/Dropbox/manuscripts/literature/bibtex/comprehensive_database.bib
csl: /Users/giulianocolosimo/Dropbox/manuscripts/literature/bibtex/apa_5th_edition.csl
citation_package: biblatex
citecolor: "red"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library("countdown")
library("plotrix")
library("latticeExtra")
library("formattable")
library("lubridate")
library("calendR")
library("formatR")

str_break = function(x, width = 65L) {
  n = nchar(x)
  if (n <= width) return(x)
  n1 = seq(1L, n, by = width)
  n2 = seq(width, n, by = width)
  if (n %% width != 0) n2 = c(n2, n)
  substring(x, n1, n2)
}
```

## Class overview

1.  Your teachers
2.  Course overview
3.  Opening quiz
4.  Bioinformatics
5.  Setting up your computers
6.  Brief intro to R and RStudio

# Your teachers

## Giuliano Colosimo

::: columns
::: {.column width="50%"}
![Dr. Giuliano Colosimo](../images/class1/giuliano_iguana.jpg){fig-alt="Dr. Giuliano Colosimo and its iguana friend" fig-align="center"}
:::

::: {.column width="50%"}
::: {style="font-size: 0.75em;"}
-   Background in Molecular Ecology
-   MS in Ecology and Evolution \@ University of Rome Tor Vergata
-   :mortar_board: in Conservation Genetics \@ Mississippi State University
-   PostDoc for 5 years \@ San Diego Zoo Wildlife Alliance
-   Research Associate working \@ University of Rome Tor Vergata
-   :envelope: giuliano.colosimo\@uniroma2.it
:::
:::
:::

## Giulia Maiello

::: columns
::: {.column width="50%"}
![Giulia Maiello](../images/class1/giulia_sterivex.jpg){fig-alt="Giulia Maiello" fig-align="center"}
:::

::: {.column width="50%"}
::: {style="font-size: 0.75em;"}
-   Background in Molecular Ecology
-   MS \@ University of Rome La Sapienza
-   :mortar_board: candidate \@ University of Rome Tor Vergata
-   Collaboration with Leeds University
-   :envelope: giulia.maiello\@uniroma2.it
:::
:::
:::

# Course overview

------------------------------------------------------------------------

::: incremental
-   Discretionary course for master level students
-   It is worth **3 CFU** for a total of 24 hours of class
-   Attendance is mandatory
-   Class will be in Italian unless foreign students are attending
-   No prior knowledge of Bioinformatics is required (but you need to know how to use a computer)
:::

. . .

| Day       | Time                          | Location   |
|-----------|-------------------------------|------------|
| Mon & Thu | 11:00 - 13:00 & 16:00 - 18:00 | Aula pesca |

------------------------------------------------------------------------

::: {style="margin-top: 250px; font-size: 1em; color: red;"}
It is a student's responsibility to periodically check the course website to make sure he/she has the most updated information about the course schedule!!
:::

------------------------------------------------------------------------

```{r calendar1, echo=FALSE, message=FALSE, warning=FALSE}
# Vector of dates
dates <- seq(as.Date("2022-10-01"), 
  as.Date("2022-10-31"), by = "1 day")

# Vector of NA and weekends
# (with the week starting on Monday)
events <- ifelse(format(dates, "%w") %in% c(6, 0), "Weekend", NA)

events[c(17, 20, 24, 27)] <- "Classes"

calendR(year = 2022, 
        month = 10, 
        start = "M",
        title = "Class calendar",
        subtitle = "October 2022",
        special.days = events,
        special.col = c("#00AAAE", "gray"),
        low.col = "white",
        legend.pos = "bottom", # Legend position
        legend.title = "",
        text = c("11:00-13:00\nMeet & Greet\nPC set up",
                 "16:00-18:00\nStudy design \nand Lab work", 
                 "11:00-13:00\nSeq. technologies",
                 "16:00-18:00\nNGS and case\n studies"),             
        text.pos = c(17, 20, 24, 27),        
        text.size = 2.5,              
        text.col = "black")
```

------------------------------------------------------------------------

```{r calendar_2, echo=FALSE, message=FALSE, warning=FALSE}
# Vector of dates
dates <- seq(as.Date("2022-11-01"), as.Date("2022-11-30"), by = "1 day")

# Vector of NA and weekends
# (with the week starting on Monday)
events <- ifelse(format(dates, "%w") %in% c(6, 0), "Weekend", NA)

events[c(3, 7, 10, 14, 17, 21, 24, 28)] <- "Classes"

calendR(year = 2022, 
        month = 11, 
        start = "M",
        title = "",
        subtitle = "November 2022",
        special.days = events,
        special.col = c("#00AAAE", "gray"),
        low.col = "white",
        legend.pos = "bottom", # Legend position
        legend.title = "",
        text = c("16:00-18:00\nPipeline 1",
                 "11:00-13:00\nPipeline 2",
                 "16:00-18:00\nPipeline 3",
                 "11:00-13:00\nPipeline 4",
                 "16:00-18:00\nAlternative\n pipelines",
                 "11:00-13:00\nEcological\n implications",
                 "16:00-18:00\nR-markdown",
                 "11:00-13:00\nRecap and\n Proj. assignment"),             
        text.pos = c(3, 7, 10, 14, 17, 21, 24, 28),        
        text.size = 2.5,              
        text.col = "black")
```

------------------------------------------------------------------------

```{r calendar_3, echo=FALSE, message=FALSE, warning=FALSE}
# Vector of dates
dates <- seq(as.Date("2022-12-01"), as.Date("2022-12-31"), by = "1 day")

# Vector of NA and weekends
# (with the week starting on Monday)
events <- ifelse(format(dates, "%w") %in% c(6, 0), "Weekend", NA)

events[23] <- "Project report due"

calendR(year = 2022, 
        month = 12, 
        start = "M",
        title = "",
        subtitle = "December 2022",
        special.days = events,
        special.col = c("Red", "gray"),
        low.col = "white",
        legend.pos = "bottom", # Legend position
        legend.title = "",
        text = c("10:00-12:00\nReport due"),             
        text.pos = 23,        
        text.size = 2.5,              
        text.col = "black")
```

# Opening quiz

------------------------------------------------------------------------

::: {style="margin-top: 150px;"}
Navigate to the following page:

<https://forms.gle/gGz2s246xyrwcPGG7>

::: callout-note
You have 15 minutes to reply to all questions on the quiz. Worry not, the quiz result will not count towards your final grade!
:::
:::

# Bioinformatics

## What is Bioinformatics?

::: {.callout-note icon="false"}
## From [Wikipedia](https://en.wikipedia.org/wiki/Bioinformatics)

Bioinformatics is an interdisciplinary field that develops methods and software tools for understanding biological data, in particular when the data sets are large and complex. \[...\] bioinformatics combines biology, chemistry, physics, computer science, information engineering, mathematics and statistics \[...\] using computational and statistical techniques.
:::

. . .

::: {.callout-important icon="false"}
Bioinformaticians deal with a variety of different type of data! **We are not bioinformaticians!!** We are evolutionary biologists!! We will focus on nucleotide sequence data trying not to lose focus of the research questions we are interested in.
:::

## Why Bioinformatics

::: incremental
-   Back in the days, biologists could analyze their data on personal computers. [Sanger Sequecing](https://en.wikipedia.org/wiki/Sanger_sequencing) produced anywhere between 50 and 1000 nucleotides at a time. It took time and resources to combine many sequences.
-   Since 1994 scientist have begun developing new technologies to parallelize DNA sequencing and produce high throughput data.
-   As these technologies improve, biologists need to harness the tools to design meaningful experiments, produce the data, and analyse them.
:::

------------------------------------------------------------------------

```{r seq_cost, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Drop of sequencing cost for Genomes and Nucleotides. Data from [https://www.genome.gov/about-genomics/fact-sheets/DNA-Sequencing-Costs-Data](https://www.genome.gov/about-genomics/fact-sheets/DNA-Sequencing-Costs-Data)"}

seq_cost <- read.csv("~/Documents/websites/abem_22-23/data/Sequencing_Cost_Data_Table_Aug2021.csv",
                     header = T)

seq_cost <- seq_cost[1:75, 1:3]

seq_cost$Cost.per.Mb <- currency(seq_cost$Cost.per.Mb, "$")
seq_cost$Cost.per.Genome <- currency(seq_cost$Cost.per.Genome, "$")

seq_cost$Date <- sub(pattern = "-", "", seq_cost$Date, fixed = T)
seq_cost$Date <- paste("01", seq_cost$Date, sep = "")
seq_cost$Date <- as.Date(seq_cost$Date, format = "%d%b%y")

seq_cost$scaledMB <- scale(seq_cost$Cost.per.Mb)
seq_cost$scaledGen <- scale(seq_cost$Cost.per.Genome)

par(mar = rep(5, 4))
plot(seq_cost$Date, seq_cost$scaledGen, type = "l", lwd = 20, col = "blue",
     ylab = "", xlab = "Time", ylim = c(-.5, 5),
     xlim = c(as.Date("2000-01-01"), as.Date("2025-01-01")),
     bty="n", yaxt = "n")
lines(seq_cost$Date, seq_cost$scaledMB, type = "l", lwd = 5, col = "red")
axis(side = 2, at = c(-.5, 2.25, 5), labels = c("$ 689", "$ 50Mln", "$ 100Mln"), las = 2, col = "blue", col.ticks = "blue", col.axis = "blue",
     lwd = 3)
axis(side = 4, at = c(-.5, 2.25, 5), labels = c("$ 0.008", "$ 2.5K", "$ 5K"), las = 2, col = "red", col.ticks = "red", col.axis = "red",
     lwd = 3)
legend("top", col = c("blue", "red"), legend = c("Cost per Genome", "Cost per Mb"),
       lty = 1, lwd = c(10, 5), bty = "n")
```

## How big is big?

. . .

```{r bp_50}
#| output-location: fragment
bp_50 <- paste(sample(c("a", "c", "t", "g"), 50, 
                      replace = T), collapse = "")
cat(bp_50)
```

. . .

```{r bp_100}
#| output-location: fragment
bp_100 <- paste(sample(c("a", "c", "t", "g"), 100,
                       replace = T), collapse = "")
cat(str_break(bp_100), sep = '\n')
```

. . .

```{r bp_500}
#| output-location: fragment
bp_500 <- paste(sample(c("a", "c", "t", "g"), 500, 
                       replace = T), collapse = "")
cat(str_break(bp_500), sep = '\n')
```

## Even bigger...

```{r bp_10000}
#| output-location: fragment
bp_10000 <- paste(sample(c("a", "c", "t", "g"),
                         10000, replace = T), collapse = "")
cat(str_break(bp_10000), sep = '\n')
```

. . .

OK, but how big?

------------------------------------------------------------------------

Genome sizes are given in either unit of mass (picograms, 1 pg = $10^{-12}$ g), or in number of base pairs (megabases, 1 Mb = $10^6$ bases). We can convert from one unit to the other using 1 pg = 978 Mb.

More information on the Eukaryotic Genome Size database [here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1669731/).

::: callout-note
## Remember...

-   1 Mb = $10^6$ = 1000000 bases
-   1 Gb = $10^9$ = 1 Mb \* $10^3$ = 1000000000 bases
:::

------------------------------------------------------------------------

```{r bp_10x6, cache=TRUE}
bp_10x6 <- paste(sample(c("a", "c", "t", "g"),
                         1e6, replace = T), collapse = "")
cat(str_break(bp_10x6), sep = '\n')
```

------------------------------------------------------------------------

![Genome size variation in amniotes [@Kapusta2017]](../images/class1/genome_size.png)

. . .

How all of this translates in computer terms?

------------------------------------------------------------------------

A single nucleotide takes approximately 1 byte (i.e., 8-bit) of memory (this is may vary based on the type of encoding used by the system, but it is a good approximation).

![Binary code scheme for nucleotides as implemented in the *ape* package [@Paradis2012]](../images/class1/bit_coding.png)

------------------------------------------------------------------------

![Bit-level coding scheme as implemented in the *ape* package [@Paradis2012]](../images/class1/bit_level_coding_scheme.png)

------------------------------------------------------------------------

Based on what we know, we can make the following approximations:

::: incremental
-   [*Homo sapiens*](http://www.genomesize.com/result_species.php?id=4420) genome (3500 Mb) ≈ 3.5 GB of data
-   [*Caenorhabditis elegans*](https://en.wikipedia.org/wiki/Caenorhabditis_elegans#Size_and_gene_content) genome (100 Mb) ≈ 100 MB of data
-   [*Polychaos dubium*](https://en.wikipedia.org/wiki/Polychaos_dubium) genome ($670*10^3$ Mb) ≈ 670 GB of data
:::

## A small example

```{r}
start_codon <- "atg"

bp_1000 <- paste(sample(c("a", "c", "t", "g"), 1000, 
                      replace = T),
               collapse = "")
start.time <- Sys.time()
bp_1000 <- gsub(start_codon, "ATG", bp_1000)
end.time <- Sys.time()
time.taken <- end.time - start.time
time.taken
```

. . .

```{r}
bp_1e6 <- paste(sample(c("a", "c", "t", "g"), 1e6, 
                        replace = T),
                 collapse = "")
start.time <- Sys.time()
bp_1e6 <- gsub(start_codon, "ATG", bp_1e6)
end.time <- Sys.time()
time.taken <- end.time - start.time
time.taken
```

# Setting up your computer

------------------------------------------------------------------------

To visualize and analyse data and to produce reports we will be using **R** and **RStudio**. These are free cross-platform software.

-   Visit the [R](https://cran.r-project.org) page and download the version compatible to your OS. Install the software following the instructions. Ideally you should install R version 4.2.0.

-   Visit the [RStudio](https://www.rstudio.com/products/rstudio/download/) page and download the desktop version compatible with your OS. Install the software following the instructions. Ideally you should install RStudio 2022.07.1+554

------------------------------------------------------------------------

::: {style="margin-top: 150px;"}
::: callout-warning
Learning all the specifics of **R** and **RStudio** is beyond the scope of this course. We will be doing a brief introduction to these tools, but it is a student's responsibility to learn how to use them in an efficient way. [Un Viaggio nell'Ambiente R](https://tommaso-russo.github.io/libroR/index.html) is a very useful resource to know how to use this tool.
:::

::: callout-tip
Remember that both teachers are available to help you figuring things out!
:::
:::

------------------------------------------------------------------------

Once installed, open **RStudio**. If everything worked appropriately when you first open **RStudio**, you should see 4 different panes organized in the following way:

::: {style="font-size: 0.75em;"}
-   the editor where you edit scripts, documents and can run code directly in the console;
-   the **R** console, where the code is run. You can type code directly into the console OR you can send it to the console by running it from the editor;
-   the environment/history panes. This is where you can see variables you've created and a full history of functions / commands you have run;
-   a fourth pane with several tabs: files/plots/packages/help/viewer. The files pane allows you to explore files on your computer like you would in Windows or Mac Finder. The plots tab is where you see the plots you create.
:::

------------------------------------------------------------------------

![R-Studio screenshot. You can change many of the aspects parameters (colors, font, panes etc.etc...).](../images/class1/rstudio.png){fig-alt="R-Studio screenshot" fig-align="center"}

# Brief intro to R and RStudio

## A *number cruncher* on steroids

```{r}
# basic arithmetic
3 + 5
```

. . .

```{r}
# assigning values to object
a <- 5
b <- 3
(c <- a + b)

```

## Basic installation

```{r sI}
sessionInfo()
```

## Packages

There are thousands of available libraries (aka, *packages*) that can be used to perform very specific analyses and produce publication-grade graphical output.

```{r packs}
sI <- sessionInfo()
sI$basePkgs
```

------------------------------------------------------------------------

```{r}
rownames(installed.packages())
```

. . .

<br/>

```{r, include=TRUE, eval=FALSE}
install.packages("ggplot2", 
                 repos = "https://cloud.r-project.org")
```

## Getting help

**R** comes with a great help system that everybody is encouraged to read extensively and consult every time they are experiencing issues when running some code or using a function.

```{r}
library(ggplot2)
```

. . .

<br/>

```{r}
?ggplot
```

---

## Packages for this course

::: callout-warning
If you have a Windows computer you need to install an additional package named *Rtools*
:::

```{r, include=TRUE, eval=FALSE}
# Only for Windows users!
install.packages("Rtools")
```

<br/>

. . .

```{r, include=TRUE, eval=FALSE}
# For everyone
source("https://raw.githubusercontent.com/GiulianoColosimo/abem_22-23/main/code/install_pks/install_packages_01.R")
```

<br/>

. . .

```{r, include=TRUE, eval=FALSE}
source("https://raw.githubusercontent.com/GiulianoColosimo/abem_22-23/main/code/install_pks/install_packages_02.R")
```

# References
